src/baseline_rule.py

```diff
@@ -121,22 +138,115 @@ def get_port_array(self):
-            port_rec = {'protocol': self.protocol} if self.protocol else {}
-            port_rec['port'] = port
-            ports_array.append(port_rec)
+        port_rec = {'protocol': self.protocol} if self.protocol else {}
+        if self.port_min == self.port_max:
+            port_rec['port'] = self.port_min
+        elif self.port_min < self.port_max:
+            port_rec['port'] = self.port_min
+            port_rec['endPort'] = self.port_max
+        ports_array.append(port_rec)
+        return ports_array
 
+    def get_port_array_calico(self):
+        """
+        :return: the port range specified in the baseline rule as a list of calico ports
+        :rtype: list
+        """
+        if not self.port_min:
+            return []
+        ports_array = []
+        if self.port_min == self.port_max:
+            ports_array = [self.port_min]
+        elif self.port_min < self.port_max:
+            ports_array = [f'{self.port_min}:{self.port_max}']
         return ports_array
 
+    def _get_calico_policy_spec_second_direction(self, is_ingress):
+        policy_spec = {'types': ['Ingress']} if is_ingress else {'types': ['Egress']}
+        if is_ingress:
+            policy_selector = self._selectors_as_netpol_peer_calico(self.target)
+            rule_dict = {'action': 'Allow',
+                         'source': self._selectors_as_netpol_peer_calico(self.source)}
+        else:
+            policy_selector = self._selectors_as_netpol_peer_calico(self.source)
+            rule_dict = {'action': 'Allow',
+                         'destination': self._selectors_as_netpol_peer_calico(self.target)}
+
+        policy_spec.update(policy_selector)
+        rule_type = 'ingress' if is_ingress else 'egress'
+        policy_spec.update({rule_type: [rule_dict]})
+        return policy_spec
+
+    def _get_policy_type(self):
+        """
+        :return: a tuple of (1) is_ingress bool flag (2) str of policy type (Ingress/Egress)
+        :rtype (bool, str)
+        """
+        is_ingress = not isinstance(self.target, IpSelector)
+        return is_ingress, 'Ingress' if is_ingress else 'Egress'
+
+    @staticmethod
+    def _get_calico_policy_dict(policy_spec, policy_name):
+        return {
+            'apiVersion': 'projectcalico.org/v3',
+            'kind': 'GlobalNetworkPolicy',
+            'metadata': {'name': policy_name},
+            'spec': policy_spec
+        }
+
+    def to_global_netpol_calico(self):
+        """
+        Create Calico GlobalNetworkPolicy resources representing the connections specified by the rule.
+        Relevant for baseline rules with no namespace, thus using GlobalNetworkPolicy which applies to all namespaces.
+        Note that two GlobalNetworkPolicy resources may be required for allowing both directions of a connection.
+
+        :return: One or two Calico GlobalNetworkPolicy resources representing the connections specified by the rule
+        :rtype: (dict, union[dict, None])
```

Comment:
After looking at the code in baseline-rules-verifier, it seems like its better if this function returns a list

Reply:
done